Chore/implement code review feedbacks

Kenec opened the pull request:
#### What does this PR do?
- implement feedbacks from Check Point Defence
#### Description of Task to be completed?
- refactor the variable and function names
- updated the github repo readme file
- refactored the api route naming convention
- remove all commented out code
#### How should this be manually tested?
- compare the PR with the master branch
- visit the application url 
#### Any background context you want to provide?
- Previous feedbacks has been implemented and merged to the master branch
#### What are the relevant pivotal tracker stories?
- None
#### Screenshots (if appropriate)
- None
#### Questions:
- None

Commit:
chore/remove .idea and selenium-debug.log file:

- remove .idea file
- remove selenium-debug.log file

[Fixes Project folder structure]

Commit:
chore(All files): Remove all commented codes

- remove all commented out codes

[Fixes All Files]

Commit:
fix(All Files): Review and modify variable and method names

- shorten lengthy variable and method names
- review and modify JSDoc

[Fixes variable and method names]

Commit:
fix(api routes): refactor the api route

- ensure that the api route follow the naming convention

[Fixes api routes]

Commit:
modifiy groupControllerTest

Commit:
remove redux dev tool from the store creator

olawalejarvis commented on client/actions/groupActions.js:
```diff
@@ -1,177 +1,153 @@
+ * @param {number} offset - the offset
+ * @return {json} -  axios post response
  */
 export function searchAllUsers(username, offset) {
```
Is a way to search with other names also - maybe through full name or email

olawalejarvis commented on client/actions/groupActions.js:
```diff
@@ -1,177 +1,153 @@
- * @return {void}
+ * @return {json} - axios post response
  */
 export function removeUserFromGroup(id, payLoad) {
```
what is payLoad?

olawalejarvis commented on client/components/GroupBoard.jsx:
```diff
@@ -36,11 +35,11 @@ class GroupBoard extends Component {
       groupId: '',
       Message: '',
-      priority_level: '',
+      priority_level: 'Normal',
```
I think this should be in camelCase

olawalejarvis commented on client/components/GroupBoard.jsx:
```diff
@@ -426,12 +391,12 @@ export default connect(mapStateToProps,
   { addNotification,
```
I think redux bindActionCreators could be used here, you can read more about to implement it

Commit:
fix(Authentication): Redirect to siginin page when token expires

- Redirect to signin page when token expires

[Fixes Authentication Page]

Commit:
write more component test

Commit:
write all components, reducers and actions test

Commit:
arrange endpoint routes into seperate files

Commit:
rename signin controller function from list to signin

Commit:
redirect user to the app on signup

Commit:
modify the status code responses of the controllers

Commit:
chore(JSDoc): refactor JSDoc for Component

- refactor JSDoc for component
- include the @extend for class component
- include @function for function component

[Fixes Component JSDoc]

Commit:
chore(Test): increase client and server test coverage

- refactored the test description messages
- write more tests to cover some controllers modifications

[Fixes Test]

Commit:
fix(GroupMember.jsx and Search.jsx):  fix eslint violations in GroupMember and Search.jsx

- replace alert and confirm with swal
- use react paginate in Search.jsx
- add top padding to the footer text

[Fixes Search.jsx]
